CMakeLists.txt

```diff
@@ -84,4 +84,8 @@ install(
     DIRECTORY ${CMAKE_SOURCE_DIR}/include/
     DESTINATION include/couchbaselitecpp
     FILES_MATCHING PATTERN "*.h*"
-)
\ No newline at end of file
+)
+
+if(CMAKE_BUILD_TYPE MATCHES Debug)
+    add_definitions(-DSHOW_DETAILED_DATABASE_MESSAGES)
```

Comment:
please use "target_compile_definition" instead. Since we only want to use this at our library (target). otherwise this command will pass this to all targets.